### PR TITLE
Screenshots of unit modals added to curric automate script

### PR DESCRIPTION
**Pull request by orangemug**
## Description
Added optional screenshots of unit modals to the curriculum automation script. 

## Issue(s)

Fixes `CUR-1058`

## How to test
Note this will take a while to run, as it's screenshotting every sequence/unit-modal

1. Clone the repo
2. Start the dev server `npm run dev`
3. `./scripts/dev/curriculum/automate screenshot :foo --modals`
4. `./scripts/dev/curriculum/automate screenshot :bar --modals`
5. `./scripts/dev/curriculum/automate screenshot compare :foo :bar`

Check the page launched from `compare` command.

**Note:** It doesn't currently render the unit options only the unit options overview page currently.

### CODE REVIEW

File: scripts/dev/curriculum/_commands/helpers.ts [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      left: 0,
+    };
+
+    images.push(def);
+    imgHeight = def.bottom;
+  }
+
+  if (images.length < 1) {
+    throw new Error("Missing");
+  }
```
Comment (assadk88) [inline scripts/dev/curriculum/_commands/helpers.ts:216]: Perhaps this error message could be a bit more descriptive?

File: scripts/dev/curriculum/_commands/helpers.ts [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    const labelName = input.replace(/^:/, "");
+    return `${BASE_PATH}/screenshots/${labelName}/`;
+  } else {
+    return input;
+  }
+}
+
+export async function readJson(filepath: string) {
+  return JSON.parse((await readFile(filepath)).toString());
+}
```
Comment (assadk88) [inline scripts/dev/curriculum/_commands/helpers.ts:102]: We could add error handling here for any promise rejects

File: scripts/dev/curriculum/_commands/helpers.ts [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+  screenshot: string;
+};
+export function renderComparison(
+  baseLabel: string,
+  targetLabel: string,
+  before: Comparison | null,
+  after: Comparison | null,
+) {
+  if (!before || !after) {
+    throw new Error("Missing");
```
Comment (assadk88) [inline scripts/dev/curriculum/_commands/helpers.ts:119]: As elsewhere, we could add a more descriptive error message here

File: scripts/dev/curriculum/_commands/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+          </div>
+        </body>
+      </html>
+    `;
+}
+
+export function resolveInputPath(input: string) {
+  if (input.startsWith(":")) {
+    const labelName = input.replace(/^:/, "");
+    return `${BASE_PATH}/screenshots/${labelName}/`;
```
Comment (assadk88) [inline scripts/dev/curriculum/_commands/helpers.ts:94]: Would it be worth using Node's path module to handle path resolution, or would that be overkill?

**Review by assadk88 — CHANGES_REQUESTED** [review]
Looks great! Just some small few things, but feel free to ignore these if they are not applicable

File: scripts/dev/curriculum/_commands/helpers.ts (hunk 3 above)
Comment (orangemug) [reply scripts/dev/curriculum/_commands/helpers.ts]: Fixed, thanks!

File: scripts/dev/curriculum/_commands/helpers.ts (hunk 4 above)
Comment (orangemug) [reply scripts/dev/curriculum/_commands/helpers.ts]: Yup changed over to use `join`

File: scripts/dev/curriculum/_commands/helpers.ts (hunk 1 above)
Comment (orangemug) [reply scripts/dev/curriculum/_commands/helpers.ts]: Fixed

**Review by assadk88 — APPROVED** [review]

**Pull request merged**